app/src/main/java/ch/epfl/sdp/peakar/user/profile/NewProfileActivity.java

```diff
@@ -305,11 +329,89 @@ public boolean dispatchTouchEvent(MotionEvent ev) {
                 .load(profileImageUrl)
                 .circleCrop()
                 .into((ImageView)findViewById(R.id.profile_picture));
     }
+
+    /**
+     * Check if the user of this profile is a friend of the authenticated user.
+     * @return true of they are friends, false otherwise.
+     */
+    private boolean isFriend() {
+        return AuthService.getInstance().getAuthAccount().getFriends().stream().anyMatch(x -> x.getUid().equals(((OtherAccount)displayedAccount).getUserID()));
+    }
+
+    /**
+     * On add friend button click
+     */
+    public void addFriendButton(View view) {
+        hideFriendButtons();
+
+        // Start a new thread that will handle the process
+        Thread addThread = new Thread(){
+            @Override
+            public void run() {
+                // Add the friend and wait for the task to end
+                ProfileOutcome addFriendOutcome = AuthService.getInstance().getAuthAccount().addFriend(((OtherAccount)displayedAccount).getUserID());
+
+                // Update the view on the UI thread
+                runOnUiThread(() -> {
+
+                    String outcomeMessage = getResources().getString(addFriendOutcome.getMessage());
+
+                    if(addFriendOutcome == ProfileOutcome.FRIEND_ADDED) {
+                        // Update UI
+                        hideUI(false, true);
+
+                        outcomeMessage = displayedAccount.getUsername() + " " + outcomeMessage;
+                    }
+
+                    // Display the message
+                    Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), outcomeMessage, Snackbar.LENGTH_LONG);
+                    snackbar.show();
+                });
+            }
+        };
+        addThread.start();
+    }
+
+    /**
+     * On remove friend button click
+     */
+    public void removeFriendButton(View view) {
+        hideFriendButtons();
+
+        // Start a new thread that will handle the process
+        Thread removeThread = new Thread(){
+            @Override
+            public void run() {
+                // Remove the friend and wait for the task to end
+                AuthService.getInstance().getAuthAccount().removeFriend(((OtherAccount)displayedAccount).getUserID());
+
+                // Update the view on the UI thread
+                runOnUiThread(() -> {
+                    // Update UI
+                    hideUI(false, false);
+
+                    String removedMessage = displayedAccount.getUsername() + " " + getResources().getString(R.string.friend_removed);
+                    // Display the message
+                    Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), removedMessage, Snackbar.LENGTH_LONG);
+                    snackbar.show();
+                });
+            }
+        };
+        removeThread.start();
+    }
+
+    /**
+     * On social activity button click
+     */
+    public void socialActivityButton(View view) {
```

Comment:
What is this?

Reply:
It is the "friends" buttons, that actually loads the `SocialActivity`.

Reply:
You mean the icon in the top right corner? I don't think the idea was that it was supposed to be a button. More an icon with different color if you are friends or not

Reply:
Oh, I thought this was a button for the `SocialActivity`. Anyway, you can see if you are friends because you have the add/remove button so it would be redundant. Personally, I like it this way. What do you think?